### DISCUSSION TITLE
Field in useFieldArray not updating onChange

### DISCUSSION THREAD

**Discussion by drydenwilliams** [react-hook-form/react-hook-form#6787] [Q&A]
I can't seem to get the fields or `pages` to update `onChange` when using `useFieldArray`. Here's my sandbox: 

[https://codesandbox.io/s/damp-fire-fcfbg](https://codesandbox.io/s/damp-fire-fcfbg?file=/src/App.js)

Editing the `siteName` and `homepageUrl` update as expected `onChange`. But, when editing the `pages` which are using `useFieldArray` it doesn't update `onChange` only when you re-submit the form. 

Any help would be amazing! Thanks in advance. 

---

**NOTE:**:
The `pages` should be a sub-page of the `homepageUrl` field and why there is a custom `test()` within `yup`.

**Comment by bluebill1049** [comment]
> it doesn't update onChange only when you re-submit the form.

do you mean the errors is not updating?
